README.md
# Laravel Algerian provinces

![Packagist Version](https://img.shields.io/packagist/v/elaborate-code/laravel-algerian-provinces?style=for-the-badge)
![Packagist Downloads](https://img.shields.io/packagist/dt/elaborate-code/laravel-algerian-provinces?style=for-the-badge)
![GitHub Workflow Status](https://img.shields.io/github/workflow/status/elaborate-code/laravel-algerian-provinces/run-tests?label=Tests&style=for-the-badge)
![GitHub Workflow Status](https://img.shields.io/github/workflow/status/elaborate-code/laravel-algerian-provinces/Fix%20PHP%20code%20style%20issues?label=Code%20Style&style=for-the-badge)

![banner](https://banners.beyondco.de/Algerian%20provinces.png?theme=dark&packageManager=composer+require&packageName=elaborate-code%2Flaravel-algerian-provinces&pattern=architect&style=style_1&description=A+model+seeded+with+the+58+Algerian+provinces+for+Laravel+apps&md=1&showWatermark=0&fontSize=100px&images=database)

[Algerian provinces](wilayas.md) Eloquent model for Laravel applications.

## Installation

Install the package via composer:

```bash
composer require medilies/laravel-algerian-provinces
```

Requirements:

1. SQLite.

### Publishing config file [Optional]

You can publish the config file with:

```bash
php artisan vendor:publish --tag="algerian-provinces-config"
```

This is the contents of the published config file:

```php
return [
    'columns_names' => [
        'fr_name' => null, // defaults to fr_name
        'ar_name' => null, // defaults to ar_name
    ],
];
```

That allows you to rename `columns` names.

## Usage

Now you have access to the `ElaborateCode\AlgerianProvinces\Models\Wilaya`

```php
use ElaborateCode\AlgerianProvinces\Models\Wilaya;

$wilayas = Wilaya::all();

$wilayas = Wilaya::find(31); // Model of: ["id" => 31, "fr_name" => "Oran", "ar_name" => "وهران"]

$wilayas = Wilaya::arr(); // Wilaya::all()->toArray()

$fr_names = Wilaya::frNames(); // Wilaya::pluck('fr_name')

$ar_names = Wilaya::arNames(); // Wilaya::pluck('ar_name')
```

```php
namespace App\Models;

use ElaborateCode\AlgerianProvinces\Models\Wilaya;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class User extends Model
{
    // use as relationship
    
    public function birthWilaya(): BelongsTo
    {
        return $this->belongsTo(Wilaya::class, 'birth_wilaya', 'fr_name');
    }
}
```

## Sushi

This package requires 0 config, and comes with no migrations or seeders thanks to the wonderful package [calebporzio/sushi](https://github.com/calebporzio/sushi). That allows us to offer a ready to use model without going the extra steps of affecting your database.

> However this dependency requires SQLite to be setup on your system!

If you prefer having a table in your database checkout [v1](https://github.com/elaborate-code/laravel-algerian-provinces/tree/v1-migration) which doesn't depand on calebporzio/sushi.

## Testing

```bash
vendor/bin/pint
```

## Changelog

Please see [CHANGELOG](CHANGELOG.md) for more information on what has changed recently.

## Contributing

Please see [CONTRIBUTING](CONTRIBUTING.md) for details.

## License

The MIT License (MIT). Please see [License File](LICENSE.md) for more information.


CHANGELOG.md
# Changelog

All notable changes to `laravel-algerian-provinces` will be documented in this file.

## v2.0.0-beta - 2022-09-01

**Full Changelog**: https://github.com/elaborate-code/laravel-algerian-provinces/compare/1.0.0...v2.0.0-beta

## 1.0.0 - 2022-07-27

### What's Changed

- Migrating and seeding by @medilies in https://github.com/elaborate-code/laravel-algerian-provinces/pull/1

### New Contributors

- @medilies made their first contribution in https://github.com/elaborate-code/laravel-algerian-provinces/pull/1

**Full Changelog**: https://github.com/elaborate-code/laravel-algerian-provinces/commits/1.0.0

## 0.2.0-beta - 2022-07-27

**Full Changelog**: https://github.com/elaborate-code/laravel-algerian-provinces/compare/0.1.0-beta...0.2.0-beta

## 0.1.0-beta - 2022-07-27

### What's Changed

- Migrating and seeding by @medilies in https://github.com/elaborate-code/laravel-algerian-provinces/pull/1

### New Contributors

- @medilies made their first contribution in https://github.com/elaborate-code/laravel-algerian-provinces/pull/1

**Full Changelog**: https://github.com/elaborate-code/laravel-algerian-provinces/commits/0.1.0-beta


arrays/wilayas.php
<?php

return  [
    [
        'id' => 1,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Adrar',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'أدرار',
    ],
    [
        'id' => 2,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Chlef',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'الشلف',
    ],
    [
        'id' => 3,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Laghouat',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'الأغواط',
    ],
    [
        'id' => 4,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Oum El Bouaghi',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'أم البواقي',
    ],
    [
        'id' => 5,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Batna',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'باتنة',
    ],
    [
        'id' => 6,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Béjaïa',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'بجاية',
    ],
    [
        'id' => 7,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Biskra',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'بسكرة',
    ],
    [
        'id' => 8,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Béchar',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'بشار',
    ],
    [
        'id' => 9,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Blida',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'البليدة',
    ],
    [
        'id' => 10,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Bouira',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'البويرة',
    ],
    [
        'id' => 11,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Tamanrasset',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'تمنراست',
    ],
    [
        'id' => 12,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Tébessa',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'تبسة',
    ],
    [
        'id' => 13,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Tlemcen',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'تلمسان',
    ],
    [
        'id' => 14,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Tiaret',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'تيارت',
    ],
    [
        'id' => 15,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Tizi Ouzou',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'تيزي وزو',
    ],
    [
        'id' => 16,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Alger',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'الجزائر',
    ],
    [
        'id' => 17,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Djelfa',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'الجلفة',
    ],
    [
        'id' => 18,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Jijel',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'جيجل',
    ],
    [
        'id' => 19,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Sétif',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'سطيف',
    ],
    [
        'id' => 20,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Saïda',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'سعيدة',
    ],
    [
        'id' => 21,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Skikda',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'سكيكدة',
    ],
    [
        'id' => 22,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Sidi Bel Abbès',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'سيدي بلعباس',
    ],
    [
        'id' => 23,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Annaba',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'عنابة',
    ],
    [
        'id' => 24,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Guelma',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'قالمة',
    ],
    [
        'id' => 25,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Constantine',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'قسنطينة',
    ],
    [
        'id' => 26,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Médéa',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'المدية',
    ],
    [
        'id' => 27,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Mostaganem',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'مستغانم',
    ],
    [
        'id' => 28,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => "M'Sila",
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'المسيلة',
    ],
    [
        'id' => 29,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Mascara',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'معسكر',
    ],
    [
        'id' => 30,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Ouargla',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'ورقلة',
    ],
    [
        'id' => 31,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Oran',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'وهران',
    ],
    [
        'id' => 32,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'El Bayadh',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'البيض',
    ],
    [
        'id' => 33,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Illizi',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'اليزي',
    ],
    [
        'id' => 34,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Bordj Bou Arreridj',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'برج بوعريريج',
    ],
    [
        'id' => 35,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Boumerdès',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'بومرداس',
    ],
    [
        'id' => 36,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'El Tarf',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'الطارف',
    ],
    [
        'id' => 37,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Tindouf',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'تندوف',
    ],
    [
        'id' => 38,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Tissemsilt',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'تسمسيلت',
    ],
    [
        'id' => 39,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'El Oued',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'الوادي',
    ],
    [
        'id' => 40,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Khenchela',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'خنشلة',
    ],
    [
        'id' => 41,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Souk Ahras',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'سوق أهراس',
    ],
    [
        'id' => 42,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Tipaza',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'تيبازة',
    ],
    [
        'id' => 43,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Mila',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'ميلة',
    ],
    [
        'id' => 44,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Aïn Defla',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'عين الدفلى',
    ],
    [
        'id' => 45,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Naâma',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'النعامة',
    ],
    [
        'id' => 46,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Aïn Témouchent',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'عين تموشنت',
    ],
    [
        'id' => 47,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Ghardaïa',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'غرداية',
    ],
    [
        'id' => 48,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Relizane',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'غليزان',
    ],
    [
        'id' => 49,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => "El M'Ghair",
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'المغير',
    ],
    [
        'id' => 50,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'El Meniaa',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'المنيعة',
    ],
    [
        'id' => 51,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Ouled Djellal',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'أولاد جلال',
    ],
    [
        'id' => 52,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Bordj Badji Mokhtar',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'برج باجي مختار',
    ],
    [
        'id' => 53,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Béni Abbès',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'بني عباس',
    ],
    [
        'id' => 54,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Timimoun',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'تيميمون',
    ],
    [
        'id' => 55,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Touggourt',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'تقرت',
    ],
    [
        'id' => 56,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'Djanet',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'جانت',
    ],
    [
        'id' => 57,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'In Salah',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'عين صالح',
    ],
    [
        'id' => 58,
        config('algerian-provinces.columns_names.fr_name') ?? 'fr_name' => 'In Guezzam',
        config('algerian-provinces.columns_names.ar_name') ?? 'ar_name' => 'عين قزّام',
    ],
];
